### DISCUSSION TITLE
Animations

### DISCUSSION THREAD

**Discussion by gucio321** [AllenDang/giu#643] [Show and tell]
Hi there!
While working on one of my projects, I decided to develop a small animations system: https://github.com/gucio321/giu-animations

**Comment by AllenDang** [comment]
Ha! It's interesting.
Maybe the syntax could be like this.
```go
g.Button(...).Animate.Color(from, to).On(trigger)
g.Button(...).Animate.Position(start, end, start).On(trigger)
```

**Reply by gucio321** [reply]
Yes, I think, I can write some api like that and open a PR against giu.

**Reply by gucio321** [reply]
@AllenDang I've just releaset v2.0.0: https://github.com/gucio321/giu-animations/tree/v2.0.0
the sintax is as follows:
```go
# to animate color of button
animations.Animator(animations.ColorFlowStyles(giu.Button("I'm a button!"), giu.StyleColorButton, giu.StyleColorButtonHovered)).Trigger(animations.TriggerOnTrue, imgui.IsItemHovered)
```

**Comment by gucio321** [comment]
I've implemented Move animation!
Let me know what do you think about it!

**Reply by AllenDang** [reply]
Cool!
